src/app/exam/submissions/ExamLiveSubmissions.js

```diff
@@ -0,0 +1,139 @@
+import { ExamInPageNavigationBar } from "../ExamInPageNavigationBar";
+import { ItemListPage } from "../../commons/ItemListPage/ItemListPage";
+import { useParams, useRouteMatch } from "react-router-dom";
+import React, { useState, useEffect } from "react";
+import { useExamContext } from "../questions/ExamContext";
+import { TableCell } from "../../../utils/TableCell";
+import { displayDate } from "../../../utils/utils";
+import FakeLink from "../../commons/FakeLink";
+import VerdictIssuedEvent from "../../../models/VerdictIssuedEvent";
+import LiveSubmissionEvent from "../../../models/LiveSubmissionEvent";
+import { studentService, problemService, liveSubmissionsService } from "../../../services/services";
+import './ExamLiveSubmissions.scss';
+
+var examSubscriptions = {};
+
+const ExamLiveSubmissions = () => {
+    const { url: currentURL } = useRouteMatch();
+    const { currentExam, refetchExam } = useExamContext();
+    const { examId } = useParams();
+    const [verdictIssuedEvents, setVerdictIssuedEvents] = useState([])
+    var events = {};
+    
+    useEffect(() => {
+        if (!currentExam) {
+            refetchExam(examId);
+        }
+        subscribeEvents();
+    });
+
+    function subscribeEvents() {
+        if (!examSubscriptions[examId]) {
+            examSubscriptions[examId] = examId;
+            liveSubmissionsService.subscribeToSubmitCompletion(examId, submitSubscriber);
+            liveSubmissionsService.subscribeToVerdictCompletion(examId, verdictSubscriber);
+        }
+    }
+
+    function submitSubscriber(message) {
+        var liveSubmissionEvent = new LiveSubmissionEvent(JSON.parse(message.body));
+        var submissionId = liveSubmissionEvent.submissionId;
+        events[submissionId] = liveSubmissionEvent;
+        updateSubmissionEventProblemTitle(submissionId, liveSubmissionEvent);
+        updateSubmissionEventStudentName(submissionId, liveSubmissionEvent);
+    }
+
+    function updateSubmissionEventProblemTitle(submissionId, liveSubmissionEvent) {
+        problemService.getProblemById(liveSubmissionEvent.problemId)
+            .then(problem => {
+                events[submissionId]["problemTitle"] = problem.title;
+                updateSubmissionEvent();
+            });
+    }
+
+    function updateSubmissionEventStudentName(submissionId, liveSubmissionEvent) {
+        studentService.getStudentById(liveSubmissionEvent.studentId)
+            .then(student => {
+                events[submissionId]["studentName"] = student.name;
+                updateSubmissionEvent();
+            });
+    }
```

Comment:
move the responsibility of "data completion" from the View into LiveSubmissionService (see my comment below)